src/local-time-element-define.ts

```diff
@@ -13,13 +13,21 @@ try {
   }
 }
 
+type JSXBaseElement = JSX.IntrinsicElements extends {span: unknown}
+  ? JSX.IntrinsicElements['span']
+  : Record<string, unknown>
```

Comment:
I'm having a really hard time reading what this does 😅 

JSXBaseElement is a the JSX element `span` _if_ JSX elements extend `{span: unknwon}`? Otherwise it's just a record of `string, unknown`?

Reply:
Yeah this is a guard that essentially checks if `JSX.InstrinsicElements.span` exists. If it doesn't exist, as you say, then the type is `Record<string, unknown>`.

In other words, if a user installs this webcomponent and doesn't have `@types/react` then this will create:

```ts
JSX.InstrinsicElements['relative-time'] = Record<string, unknown> & Partial<Omit<RelativeTimeElement, keyof HTMLElement>>
```

If they _do_ have `@types/react` then this will create:

```ts
JSX.InstrinsicElements['relative-time'] = JSX.InstrinsicElements.span & Partial<Omit<RelativeTimeElement, keyof HTMLElement>>
```